library/ssl_misc.h

```diff
@@ -2313,20 +2313,20 @@ static inline int mbedtls_ssl_tls13_sig_alg_for_cert_verify_is_supported(
     const uint16_t sig_alg)
 {
     switch (sig_alg) {
-#if defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_EPHEMERAL_ENABLED)
-#if defined(PSA_WANT_ALG_SHA_256) && defined(MBEDTLS_ECP_DP_SECP256R1_ENABLED)
+#if defined(MBEDTLS_PK_CAN_ECDSA_SOME)
```

Comment:
In 9f27b4b379 the guards here change from `defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_EPHEMERAL_ENABLED` to `defined(MBEDTLS_PK_CAN_ECDSA_SOME)`. The new code looks right, but there's no indication of what's happening in the commit message. `defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_EPHEMERAL_ENABLED` was added in https://github.com/Mbed-TLS/mbedtls/pull/7921, which just got merged, and this looks like a mistake in #7921. Should we be looking for other mistakes? Why didn't review or testing catch this: is the change from #7921 correct after all?

Reply:
Yes, the change was intentional as I noticed the issue while resolving conflicts, and I meant to add a note in the commit message but missed it the first time and forgot to came back for it.

I was planning on creating a follow-up issue for 7921 because I'm not fully happy with the naming scheme for the new macros, I'll either add this to that issue or create two follow-up issues. (I've added this to my personal notes and will create the issue(s) shortly.)

Reply:
Aw, the frustrating part is I did notice this during my initial review of #7921, but then incorrectly marked as resolved later... https://github.com/Mbed-TLS/mbedtls/pull/7921/files#r1281542395

Reply:
Tracked as #8258 in the cleanups EPIC. Marking this discussion as resolved.